src/FGenerators.jl

```diff
@@ -300,12 +301,14 @@ function define_foldl(__module__::Module, funcname, structname, allargs, body)
             $completion
         end
     end
-    return expand_with_yield_defs(
-        __module__,
-        ex;
-        on_yield = _on_yield,
-        on_yieldfrom = _on_yieldfrom,
-    )
+    with_extra_state_variables([ACC]) do
+        return expand_with_yield_defs(
+            __module__,
+            ex;
+            on_yield = _on_yield,
+            on_yieldfrom = _on_yieldfrom,
+        )
+    end
```

Comment:
This PR adds `EXTRA_STATE_VARIABLES[]` to the list of state variables to be used.  This is used via https://github.com/JuliaFolds/FGenerators.jl/pull/7